warp10/build.gradle

```diff
@@ -43,171 +42,147 @@ dependencies {
+    api 'org.fusesource.leveldbjni:leveldbjni-all:1.8'
     // https://mvnrepository.com/artifact/org.openlabtesting.leveldbjni/leveldbjni-linux64-aarch64
-    compile group: 'org.openlabtesting.leveldbjni', name: 'leveldbjni-linux64-aarch64', version: '1.8'
+    api group: 'org.openlabtesting.leveldbjni', name: 'leveldbjni-linux64-aarch64', version: '1.8'
 
-    compile 'org.iq80.leveldb:leveldb:0.12:uber'
+    api 'org.iq80.leveldb:leveldb:0.12:uber'
 
     //
     // Worf
     //
-    compile 'jline:jline:2.13'
+    api 'jline:jline:2.13'
 
     //
     // Mustache
     //
-    compile 'com.github.spullara.mustache.java:compiler:0.9.1'
+    api 'com.github.spullara.mustache.java:compiler:0.9.1'
 
     //
     // Geo
     //
-    compile 'com.geoxp:geoxplib:1.0.0'
-    compile("org.wololo:jts2geojson:0.10.0")
+    api 'com.geoxp:geoxplib:1.0.0'
+    api 'org.wololo:jts2geojson:0.10.0'
 
     //
     // Java Merge Sort
     //
-    compile 'com.fasterxml.util:java-merge-sort:1.0.0'
+    api 'com.fasterxml.util:java-merge-sort:1.0.0'
 
     // @see http://www.mail-archive.com/dev@kafka.apache.org/msg03829.html
-    compile("org.apache.kafka:kafka_2.11:0.8.2.2") {
+    api('org.apache.kafka:kafka_2.11:0.8.2.2') {
         exclude module: 'jms'
         exclude module: 'jmxtools'
         exclude module: 'jmxri'
     }
-    testCompile 'junit:junit:4.6'
-}
-
-jar {
-    zip64 = true
-
-    manifest {
-        attributes(
-                "Implementation-Title": "Warp 10",
-                "Implementation-Vendor": "SenX",
-                "Implementation-Version": version)
-    }
+    testCompileOnly 'junit:junit:4.6'
 }
 
 task pack(type: Jar) {
     zip64 = true
-    ext.warpSuffixName = null != System.getProperty('warp') ? System.getProperty('warp') : ""
-    appendix = "${warpSuffixName}"
+    archiveAppendix = (null != System.getProperty('warp')) ? System.getProperty('warp') : ""
 
     manifest {
-        attributes("Main-Class": "io.warp10.standalone.Warp")
-        attributes("Implementation-Title": "Warp 10")
-        attributes("Implementation-Version": version)
+        attributes(
+            "Main-Class": "io.warp10.standalone.Warp",
+            "Implementation-Title": "Warp 10",
+            "Implementation-Vendor": "Warp 10",
+            "Implementation-Version": project.version
+        )
     }
 
     from files(sourceSets.main.output.classesDirs)
     from files(sourceSets.main.resources.srcDirs)
     from {
-        configurations.compile.collect {
+        configurations.compileClasspath.collect {
             it.isDirectory() ? it : zipTree(it).matching {
                 exclude { it.toString().contains('/servlet-api') }
-                exclude { it.toString().contains('jmh') }
                 exclude { it.toString().contains('junit') }
-                exclude { it.toString().contains('parquet') }
             }
         }
     } {
         exclude 'META-INF/*.DSA'
```

Comment:
META-INF/*.RSA signatures may appear/disappear depending on dependencies. It is important that those files be removed otherwise the JVM may fail to validate repackaged classes if they initially had a signature.